feature/platform/transceiver/otg_tests/zr_laser_bias_current_test/zr_laser_bias_current_test.go

```diff
@@ -0,0 +1,219 @@
+	fptest.RunTests(m)
+}
+
+// Topology:
+//   dut:port1 <--> port2:dut
+//
+
+func verifyLaserBiasCurrent(t *testing.T, dut1 *ondatra.DUTDevice, pStream *samplestream.SampleStream[float64], sensorName string) {
+	laserBias := pStream.Next(t)
+	if laserBias == nil {
+		t.Fatalf("laserBias telemetry %q  was not streamed in the most recent subscription interval", sensorName)
+	}
+	laserBiasVal, ok := laserBias.Val()
+	if !ok {
+		t.Fatalf("LaserBias %q telemetry is not present", sensorName)
+	}
+	if reflect.TypeOf(laserBiasVal).Kind() != reflect.Float64 {
+		t.Errorf("Return value is not type float64")
+	}
+	if laserBiasVal <= 0 && laserBiasVal >= 131 {
+		t.Errorf("The laser bias value is not between 0 and 131")
+	}
+	t.Logf("laserBias value: %f", laserBiasVal)
+}
+func interfaceConfig(t *testing.T, dut1 *ondatra.DUTDevice, dp *ondatra.Port) {
+	d := &oc.Root{}
+	i := d.GetOrCreateInterface(dp.Name())
+	i.Enabled = ygot.Bool(true)
+	i.Type = oc.IETFInterfaces_InterfaceType_ethernetCsmacd
+	gnmi.Replace(t, dut1, gnmi.OC().Interface(dp.Name()).Config(), i)
+	OCcomponent := opticalChannelComponentFromPort(t, dut1, dp)
+	gnmi.Replace(t, dut1, gnmi.OC().Component(OCcomponent).OpticalChannel().Config(), &oc.Component_OpticalChannel{
+		TargetOutputPower: ygot.Float64(targetOutputPower),
+		Frequency:         ygot.Uint64(frequency),
+	})
+}
+func TestZRLaserBiasCurrentState(t *testing.T) {
+	dut1 := ondatra.DUT(t, "dut")
+	dp1 := dut1.Port(t, "port1")
+	dp2 := dut1.Port(t, "port2")
+	t.Logf("dut1: %v", dut1)
+	t.Logf("dut1 dp1 name: %v", dp1.Name())
+	interfaceConfig(t, dut1, dp1)
+	interfaceConfig(t, dut1, dp2)
+	intUpdateTime := 2 * time.Minute
+	gnmi.Await(t, dut1, gnmi.OC().Interface(dp1.Name()).OperStatus().State(), intUpdateTime, oc.Interface_OperStatus_UP)
+	transceiverState := gnmi.Get(t, dut1, gnmi.OC().Interface(dp1.Name()).Transceiver().State())
+	if dp1.PMD() != ondatra.PMD400GBASEZR {
+		t.Fatalf("%s Transceiver is not 400ZR its of type: %v", transceiverState, dp1.PMD())
+	}
+	OCcomponent := opticalChannelComponentFromPort(t, dut1, dp1)
+	component1 := gnmi.OC().Component(OCcomponent)
+	p1StreamInstant := samplestream.New(t, dut1, component1.OpticalChannel().LaserBiasCurrent().Instant().State(), 10*time.Second)
+	verifyLaserBiasCurrent(t, dut1, p1StreamInstant, "laserbiasInstant")
+	defer p1StreamInstant.Close()
+	p1StreamMin := samplestream.New(t, dut1, component1.OpticalChannel().LaserBiasCurrent().Min().State(), 10*time.Second)
+	verifyLaserBiasCurrent(t, dut1, p1StreamMin, "laserbiasMin")
+	defer p1StreamMin.Close()
+	p1StreamMax := samplestream.New(t, dut1, component1.OpticalChannel().LaserBiasCurrent().Max().State(), 10*time.Second)
+	verifyLaserBiasCurrent(t, dut1, p1StreamMax, "laserbiasMax")
+	defer p1StreamMax.Close()
+	p1StreamAvg := samplestream.New(t, dut1, component1.OpticalChannel().LaserBiasCurrent().Avg().State(), 10*time.Second)
+	verifyLaserBiasCurrent(t, dut1, p1StreamAvg, "laserbiasAvg")
+	defer p1StreamAvg.Close()
```

Comment:
can you also do a comparison to make sure:

min <= avg <= max

It seems that everywhere you run verifyLaserBiasCurrent() it does this four times. So you could instead create a wrapper func that checks all four items and additionally does the check mentioned in this comment. That should remove a fair amount of similar code.

Reply:
Hi Eric,

Added the check and created wrapper function to reduce additional call as suggested,

Thanks